### DISCUSSION TITLE
Set workspace name without changing the workspace number

### DISCUSSION THREAD

**Discussion by Theraa0** [i3/i3#4938] [Q&A]
I am trying to change a workspace name to display its name in my polybar instead of the workspace number. The issue is that i depend on using the workspaces number in a few scripts and i would have to rewrite a bunch of lines in my config if i were to drop the number. ( I use something like `i3-msg -t get_workspaces | jq '. | map(.num)'`to get the workspaces, that is the workspace number i am talking about. )

I define the workspace using `set $ws0 "0: music"` and show its name in polybar using `label-focused = %name%` and `label-unfocused = %name%`.

This however is quite ugly as my polybar then displays my workspaces as follows:

    `What it looks like:
    0: music   1   2   3 etc.
    What the goal is:
    music   1   2   4 etc.`

Removing the ": " works as well, i want to drop the 0 too though because that looks even worse. After reading some articles and messing arround a bit i was unable to find any way to do that, can anyone here help me out? Thanks

**Comment by RayZ0rr** [comment]
You can do something like the following. Not exactly what you want but use it if you like it.

```sh
ws-icon-0 = "1; "
ws-icon-1 = "2; "
ws-icon-2 = "3;﬑ "
ws-icon-3 = "4; "
ws-icon-4 = "5; "
ws-icon-5 = "6; "
ws-icon-6 = "7; "
ws-icon-7 = "8; "
ws-icon-8 = "9; "
ws-icon-9 = "10; "

label-focused = %index%:%icon%
label-unfocused = %index%:%icon%
label-visible = %index%:%icon%
label-urgent = %index% !
```
